File: src/menus/sessions.js
/**
 * Sessions, On Deck, and Recently Added menus.
 */

import inquirer from 'inquirer';
import { sessionTable, mediaTable, printError, printInfo, printHeader } from '../display.js';
import { runMediaMenu } from './media.js';

export async function runSessionsMenu(client) {
  printHeader('Active Sessions');
  let sessions;
  try {
    sessions = await client.sessions();
  } catch (err) {
    printError(err.message);
    return;
  }
  if (!sessions.length) {
    printInfo('No active sessions.');
    return;
  }
  sessionTable(sessions);
  await inquirer.prompt([{ type: 'input', name: '_', message: 'Press Enter to return…' }]);
}

export async function runOnDeckMenu(client) {
  printHeader('On Deck');
  let items;
  try {
    items = await client.onDeck();
  } catch (err) {
    printError(err.message);
    return;
  }
  if (!items.length) {
    printInfo('Nothing on deck.');
    return;
  }
  mediaTable(items, 'On Deck');

  const { chosen } = await inquirer.prompt([
    {
      type: 'list',
      name: 'chosen',
      message: 'Select an item:',
      choices: [
        ...items.map((it) => ({
          name: [it.grandparentTitle, it.title].filter(Boolean).join(' – ') || '?',
          value: it,
        })),
        { name: '← Back', value: null },
      ],
    },
  ]);
  if (chosen) await runMediaMenu(client, chosen);
}

export async function runRecentlyAddedMenu(client) {
  printHeader('Recently Added');
  let items;
  try {
    items = await client.recentlyAdded();
  } catch (err) {
    printError(err.message);
    return;
  }
  if (!items.length) {
    printInfo('Nothing recently added.');
    return;
  }
  mediaTable(items, 'Recently Added');

  const { chosen } = await inquirer.prompt([
    {
      type: 'list',
      name: 'chosen',
      message: 'Select an item:',
      choices: [
        ...items.map((it) => ({ name: it.title ?? '?', value: it })),
        { name: '← Back', value: null },
      ],
    },
  ]);
  if (chosen) await runMediaMenu(client, chosen);
}

File: plex_cli/menus/sessions.py
"""Sessions and recently-added / on-deck menus."""

from __future__ import annotations

import questionary

from plex_cli.api import PlexAPIError, PlexClient
from plex_cli.display import console, error, header, info, media_table, session_table
from plex_cli.menus.media import run_media_menu


def run_sessions_menu(client: PlexClient) -> None:
    """Show active sessions on the server."""
    header("Active Sessions")
    try:
        sessions = client.sessions()
    except PlexAPIError as exc:
        error(str(exc))
        return

    if not sessions:
        info("No active sessions.")
        return

    console.print(session_table(sessions))
    questionary.press_any_key_to_continue("Press any key to return…").ask()


def run_on_deck_menu(client: PlexClient) -> None:
    """Show On Deck items (continue watching)."""
    header("On Deck")
    try:
        items = client.on_deck()
    except PlexAPIError as exc:
        error(str(exc))
        return

    if not items:
        info("Nothing on deck.")
        return

    console.print(media_table(items, title="On Deck"))

    choices = [
        questionary.Choice(
            f"{it.get('grandparentTitle','')} – {it.get('title','?')}".lstrip(" – "),
            value=it,
        )
        for it in items
    ]
    choices.append(questionary.Choice("← Back", value=None))
    chosen = questionary.select("Select an item:", choices=choices).ask()
    if chosen:
        run_media_menu(client, chosen)


def run_recently_added_menu(client: PlexClient) -> None:
    """Show recently added items."""
    header("Recently Added")
    try:
        items = client.recently_added()
    except PlexAPIError as exc:
        error(str(exc))
        return

    if not items:
        info("Nothing recently added.")
        return

    console.print(media_table(items, title="Recently Added"))

    choices = [questionary.Choice(it.get("title", "?"), value=it) for it in items]
    choices.append(questionary.Choice("← Back", value=None))
    chosen = questionary.select("Select an item:", choices=choices).ask()
    if chosen:
        run_media_menu(client, chosen)
